ocaml/xapi/xapi_udhcpd.ml

```diff
@@ -162,7 +162,18 @@ let restart_nolock () =
   let pid = try Unixext.pidfile_read !Xapi_globs.udhcpd_pidfile with _ -> None in
   Opt.iter Unixext.kill_and_wait pid;
   let (_: string * string) = execute_command_get_output !Xapi_globs.busybox [ "udhcpd"; !Xapi_globs.udhcpd_conf ] in
-  ()
+  let start = Mtime_clock.elapsed_ns () in
+  let rec wait_for_pid n =
+    let now = Mtime_clock.elapsed_ns () in
+    if Int64.sub now start > Int64.mul 30L 1000000000L then failwith "Failed to start udhcpd";
```

Comment:
You can use http://erratique.ch/software/mtime/doc/Mtime_clock.html#counters and compare spans

Reply:
:+1: